checksec.cpp

```diff
@@ -63,6 +64,35 @@ Checksec::Checksec(std::string filepath) : filepath_(filepath), loadedImage_(fil
         loadConfigSecurityCookie_ = loadConfig.SecurityCookie;
         loadConfigSEHandlerTable_ = loadConfig.SEHandlerTable;
         loadConfigSEHandlerCount_ = loadConfig.SEHandlerCount;
+
+        // Iterate over debug directories
+        if (!peparse::GetDataDirectoryEntry(loadedImage_.get(), peparse::DIR_DEBUG,
+                                            debugDirectories)) {
+            std::cerr << "Warn: No debug directories"
+                      << "\n";
+            return;
+        }
+
+        peparse::debug_dir_entry debugDir{};
+        uint32_t numberOfDebugDirs = debugDirectories.size() / sizeof(debugDir);
+        auto debugDirSize = std::min(debugDirectories.size(), sizeof(debugDir));
+
+        for (int i = 0; i < numberOfDebugDirs; i++) {
+            // copy the current debug directory into debugDir
+            memcpy(&debugDir, debugDirectories.data() + (i * debugDirSize), debugDirSize);
+            // 20 == IMAGE_DEBUG_TYPE_EX_DLLCHARACTERISTICS
+            // For now we only care about this debug directory since it contains the CETCOMPAT bit
+            if (debugDir.Type == 20) {
+                if (debugDir.PointerToRawData != 0) {
+                    auto dataPtr = debugDir.PointerToRawData + loadedImage_.get()->fileBuffer->buf;
+                    try {
+                        extendedDllCharacteristics_ = *((uint16_t*)dataPtr);
+                    } catch (...) {
+                        extendedDllCharacteristics_ = 0;
+                    }
```

Comment:
I might be wrong, but I think this `try...catch` isn't doing anything -- the only thing that can really happen here is a fault if `dataPtr` isn't a valid address, but `catch` won't handle that. 

I think instead we want to make sure `dataPtr` is valid within the `fileBuffer`, and remove the block.

Reply:
Fixed!

Reply:
LGTM, thanks!